reconcile/templating/test.py

```diff
@@ -0,0 +1,97 @@
+import logging
+from difflib import context_diff
+from typing import Callable, Optional
+
+from pydantic import BaseModel
+from ruamel import yaml
+
+from reconcile.gql_definitions.templating.templates import TemplateV1, query
+from reconcile.templating.rendering import TemplateData, create_renderer
+from reconcile.utils import gql
+from reconcile.utils.runtime.integration import QontractReconcileIntegration
+
+QONTRACT_INTEGRATION = "templating-test"
+
+
+def get_templates(
+    query_func: Optional[Callable] = None,
+) -> list[TemplateV1]:
+    if not query_func:
+        query_func = gql.get_api().query
+    return query(query_func).template_v1 or []
+
+
+class TemplateDiff(BaseModel):
+    template: str
+    test: str
+    diff: str
+
+
+class TemplatingTestIntegration(QontractReconcileIntegration):
+    diffs: list[TemplateDiff]
```

Comment:
No need to declare this variable here.
```
self.diffs: list[TemplateDiff] = []
```
in line 47 would be sufficient

Reply:
No, it will raise an exception. I moved it to `__init__`